client/src/extension.ts

```diff
@@ -176,7 +176,11 @@ export async function activate (context: vscode.ExtensionContext): Promise<void>
 }
 
 export function deactivate (): Thenable<void> | undefined {
-  taskProvider.dispose();
-  (logger.outputChannel as vscode.OutputChannel).dispose()
+  // The server has to be handled before the disposables.
+  // Otherwise it might attempt to use anything that has been disposed.
   return deactivateLanguageServer(client)
+    .then(() => {

```

Comment:
`await` would probably look simpler here (just my opinion). The current syntax seems like deactivateLanguageServer has something important to return but it is void.

Reply:
This is the first thing I tried. Somehow it did not work. I guess `Thenable` and `Promise` have some subtle difference I don't fully understand.